### PR TITLE
Translate `cut()`

### LINKED ISSUE #697: support for cut() function in dbplyr (snowflake)
Hi, 

I'm using dbplyr to interface with Snowflake, and a very common task is to group numeric columns by bands to summarise output. With classic dplyr one would do group_by(cut(factor, c(1, 2, 3))) and it works wonders, but dbplyr does not appear to support it (at least in snowflake). 

So this is a post of two halves:
1) Could we please build in support for cut() function in SQL translations?
2) In the mean time, what would be the best workaround for this? 

e.g. i can build a function that creates an appropriate sql phrase, but how do I get it to be evaluated properly?

```
db_cut <- function(fact, bins) {
  k = length(bins)
  string = paste0("CASE WHEN ", fact, " < ", bins[1], " THEN '<", bins[1], "' ")   
  for (i in 2:k) {
    string = paste(string,
                   paste0("WHEN ", fact, " < ", bins[i], " THEN '[",
                          bins[i-1], "-", bins[i], ")' "))}
  string = paste0(string, " END")
  return(string)
}

this does not work...:
data %>%
  mutate(bands = db_cut(column_name, c(1, 2, 3))) %>%
  group_by(bands) %>%
  summarise(n()) 

```

am I missing something simple, or going down the wrong path?

Thanks!

### PR DESCRIPTION
**Pull request by mgirlich**
Closes #697

### CODE REVIEW

File: R/translate-sql-cut.R [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  breaks <- sort.int(as.double(breaks))
+  if (anyDuplicated(breaks)) {
+    abort("`breaks` are not unique.")
+  }
+  n <- length(breaks)
+  if (n == 1) {
+    abort("`breaks` must have size 2 or bigger.")
+  }
+
+  labels <- labels %||% TRUE
```
Comment (hadley) [inline R/translate-sql-cut.R:11]: Should allow labels to be a character vector too:

```R
cut(1:3, breaks = c(0, 1.5, 2.5), labels = c("a", "b"))
```

File: tests/testthat/test-translate-sql-cut.R [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+  expect_equal(
+    translate_sql(cut(x, c(0, 1, Inf))),
+    sql_lines(
+      "CASE",
+      "WHEN (`x` <= 0.0) THEN NULL",
+      "WHEN (`x` > 0.0 AND `x` <= 1.0) THEN '(0,1]'",
+      "WHEN (`x` > 1.0) THEN '(1,Inf]'",
+      "END"
+    )
+  )
```
Comment (hadley) [inline tests/testthat/test-translate-sql-cut.R:78]: Do these hit different code paths to the case with -Inf and Inf limits? If so, I think you could drop them.

File: R/translate-sql-cut.R [hunk 3]
```diff
@@ -0,0 +1,61 @@
+sql_cut <- function(x, breaks, labels = NULL, right = TRUE) {
```
Comment (hadley) [inline R/translate-sql-cut.R:1]: Worth implementing `include.lowest`?

File: R/translate-sql-cut.R [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  breaks <- sort.int(as.double(breaks))
```
Comment (hadley) [inline R/translate-sql-cut.R:2]: This has the side-effects of stripping any `NA` from `breaks`? Is that deliberate? If so, should add a test.

**Review by hadley — APPROVED** [review]

File: tests/testthat/test-translate-sql-cut.R (hunk 2 above)
Comment (mgirlich) [reply tests/testthat/test-translate-sql-cut.R:78]: 👍

File: R/translate-sql-cut.R (hunk 4 above)
Comment (mgirlich) [reply R/translate-sql-cut.R:2]: It is what `base::cut()` does but it looks terrible to me... So I'm not completely sure it is a good idea to also do that?

File: R/translate-sql-cut.R (hunk 3 above)
Comment (mgirlich) [reply R/translate-sql-cut.R]: The case work was a bit annoying but it makes sense after all I guess.

File: R/translate-sql-cut.R (hunk 4 above)
Comment (hadley) [reply R/translate-sql-cut.R:2]: If you don't have any strong feelings, I'd suggest adding `na.last = TRUE` to `sort.int` and then erroring if `breaks` contains any NAs. I implemented something similar for scales very recently: https://github.com/r-lib/scales/blob/main/R/label-number.r#L323-L332

**Review by hadley — APPROVED** [review]

**Comment by hadley** [comment]
BTW I fixed the build failure on macOS-latest

**Pull request merged**